### ISSUE TITLE
Unable to reset local cluster

### ISSUE THREAD

**Issue by rajucomp** [Azure/service-fabric-issues#1509]
I have been trying to reset the local cluster but to no success. So we decided to manually run the powershell scripts that are employed in resetting the local cluster. We identified these scripts as "ClusterSetupUtilities" and "DevClusterSetup". We opened VS Code as Administrator and stared debugging the "DevClusterSetup" file where we land to the function "RemoveFolder" that checks whether the directories "C:\SfDevCluster\Data" and "C:\SfDevCluster\Log" exist and, if they do, delete them via this cmd command snippet "cmd /c rmdir /S /Q $folderPath > $null". However this is the exact step where we are facing issues. We always get an exception that says 
    `Program 'cmd.exe' failed to run: Access is deniedAt line:1 char:1
+ cmd /?
+ ~~~~~~.
At line:1 char:1
+ cmd /?
+ ~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [], ApplicationFailedException
    + FullyQualifiedErrorId : NativeCommandFailed`

Here is a bit more detailed exception :-

`Exception             : System.Management.Automation.ApplicationFailedException: Program 'cmd.exe' failed to run: Access is deniedAt C:\Program
                        Files\Microsoft SDKs\Service Fabric\Tools\Scripts\ClusterSetupUtilities.psm1:76 char:9
                        +         cmd /c rmdir /S /Q $folderPath > $null
                        +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~. ---> System.ComponentModel.Win32Exception: Access is denied
                           at System.Diagnostics.Process.StartWithShellExecuteEx(ProcessStartInfo startInfo)
                           at System.Management.Automation.NativeCommandProcessor.Complete()
                           --- End of inner exception stack trace ---
                           at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
                           at System.Management.Automation.Interpreter.ActionCallInstruction `2.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
                           at System.Management.Automation.DlrScriptCommandProcessor.RunClause(Action`1 clause, Object dollarUnderbar, Object
                        inputToProcess)
                        --- End of stack trace from previous location where exception was thrown ---
                           at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                           at System.Management.Automation.Internal.PipelineProcessor.SynchronousExecuteEnumerate(Object input)
                           at System.Management.Automation.PipelineOps.InvokePipeline(Object input, Boolean ignoreInput, CommandParameterInternal[][]
                        pipeElements, CommandBaseAst[] pipeElementAsts, CommandRedirection[][] commandRedirections, FunctionContext funcContext)
                           at System.Management.Automation.Interpreter.ActionCallInstruction`6.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
TargetObject          :
CategoryInfo          : ResourceUnavailable: (:) [], ApplicationFailedException
FullyQualifiedErrorId : NativeCommandFailed
ErrorDetails          :
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at RemoveFolder, C:\Program Files\Microsoft SDKs\Service Fabric\Tools\Scripts\ClusterSetupUtilities.psm1: line 76
                        at EnsureDirectoryCleaned, C:\Program Files\Microsoft SDKs\Service Fabric\Tools\Scripts\ClusterSetupUtilities.psm1: line 376
                        at SetupDataAndLogRoot, C:\Program Files\Microsoft SDKs\Service Fabric\Tools\Scripts\ClusterSetupUtilities.psm1: line 475
                        at <ScriptBlock>, C:\Program Files\Microsoft SDKs\Service Fabric\ClusterSetup\DevClusterSetup.ps1: line 94
PipelineIterationInfo : {}
PSMessageDetails      :

`

We tried commenting the error-generating command and replaced it with  the equivalent commands in powershell for removing directories like this "Remove-Item -LiteralPath $folderPath -Force -Recurse > $null". However , we are still getting this exception.  We are trying to troubleshoot this problem for the past three days but to no avail.

Any help would be greatly appreciated.

**Comment by jdvor** [comment]
Maybe it's just a file-system ACL issue.
- Have you tried running you scripts as admin?
- If admin does not help, you should check [owner](https://www.windowscentral.com/how-take-ownership-files-and-folders-windows-10) and permissions for those directories or at this point I would probably re-install whole SDK, because if Service Fabric Local Cluster Manager does not work (remove, reset, setup, ...) you may have other problems as well.

**Comment by vshan** [comment]
Hi @rajucomp , have the above suggestions mitigated your issue?

**Comment by rajucomp** [comment]
Hi @vshan, sorry for the late reply. I ran the scripts as admin and checked owner and permissions but could not get success. For a temporary workaround, we had to manually delete the folders "C:\SfDevCluster\Data" and "C:\SfDevCluster\Log" and then ran the scripts successfully. This is weird considering we can manually delete the folders but not in the scripts that we run as admin. Anyways, we really appreciate the community for their support and development. Thanks !
